backend/src/database/migrations/20210429045222-create-is-credential-used-function.js

```diff
@@ -17,13 +17,32 @@ module.exports = {
     AND j.status NOT IN ('READY','LOGGED')
     --  but they are not this campaign
     AND j.campaign_id <> campaignId
+    AND c1.type <> 'EMAIL'
     LIMIT 1;
+
+    SELECT count(*) INTO count_email_campaign_using
+    FROM campaigns c1 WHERE 
+    c1.id <> campaignId
+    AND c1.cred_name = (SELECT c.cred_name FROM campaigns c WHERE c.id = campaignId LIMIT 1)
+    AND c1.type = 'EMAIL';
+
+    SELECT
```

Comment:
to summarise, `result` returns `TRUE` if:
- This credential is being used for 2 or more email campaigns; OR (is the >2 necessary? hmm)
- This credential is being used for another non-email campaign

(would it be helpful to include the above as comments within the SQL?)

Reply:
> is the >2 necessary?

What do you mean?

> would it be helpful to include the above as comments within the SQL?

There're tests so that's better than comments

Reply:
> is the >2 necessary?

Oh I mean will there be instances where a credential is used for 3 email campaigns? Probably not if the code is working properly haha. Just thinking out loud.

Reply:
> Oh I mean will there be instances where a credential is used for 3 email campaigns? Probably not if the code is working properly haha. Just thinking out loud.

Under the right condition (i.e. 2 checks are done at the same time with 1 existing job), yes. But this is super unlikely to happen